src/main/java/uk/gov/pay/directdebit/mandate/params/MandateSearchParams.java

```diff
@@ -101,18 +107,47 @@ public MandateSearchParams withToDate(ZonedDateTime toDate) {
         return this;
     }
 
-    public MandateSearchParams withPage(int page) {
-        this.page = page;
+    public MandateSearchParams withPage(Integer page) {
+        if (page != null && page < 1) {
+            throw new BadRequestException("Query param 'page' should be a non zero positive integer.");
```

Comment:
Is it possible to reuse the existing `NegativeSearchParamException` here?

Reply:
So I didn't use `NegativeSearchParamException` because it had a pre-defined message of `Query param '%s' should be a non zero positive integer` whereas I wanted `Query param 'display_size' should be between 1 and 500.` I don't know what the benefit of using `NegativeSearchParamException` is anyway - what's wrong with throwing a `BadRequestException`? The error message is going to get logged so from the point of view of support it makes no difference to me.